### PR TITLE
V1 of old-to-new CTFO DB conversion.

### PR DESCRIPTION
**Pull request by dkorolev**
Hi @mzhurovich ,

Managed to do today:
1. Convert `user` and `card` `Dictionary` fields, `*Updated` only (no `*Deleted)`.
2. REST on top of it, hypermedia and advanced hypermedia.

The rest is still TODO. Being sick sucks.

Thanks,
Dima

### CODE REVIEW

File: convert_old_db/convert.cc [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+#include "../../Current/Bricks/dflags/dflags.h"
+#include "../../Current/Bricks/time/chrono.h"
+
+#include "../../Current/TypeSystem/struct.h"
+#include "../../Current/TypeSystem/Serialization/json.h"
+
+DEFINE_string(input, "db.json", "The name of the input data to convert.");
+DEFINE_string(output, "new_db.json", "The name of the input data to convert.");
+
+// TODO(dkorolev) + TODO(mzhurovich): Should we expose `T_FIELDS_VARIANT` or `T_TRANSACTION` from Storage?
```
Comment (mzhurovich) [inline convert_old_db/convert.cc]: `T_TRANSACTION` is public in Storage and `T_FIELDS_VARIANT` is private. We could make it public as well.

File: convert_old_db/gen_golden_db.cc [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      user.uid = static_cast<UID>(42);
+      user.level = 1;
+      user.score = 9001;
+      fields.user.Add(user);
+    }).Go();
+
+    db.Transaction([](MutableFields<DB> fields) {
+      Card card;
+      card.cid = static_cast<CID>(42);
+      card.text = "В лесу родилась ёлочка";
```
Comment (mzhurovich) [inline convert_old_db/gen_golden_db.cc:64]: )))

File: convert_old_db/new_schema.h [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  CURRENT_FIELD(uid, UID, UID::INVALID_USER);
+  CURRENT_FIELD(level, uint8_t, 0u);           // User level [0, 9].
+  CURRENT_FIELD(score, uint64_t, score = 0u);  // User score.
+
+  UID key() const { return uid; }
+  void set_key(const UID value) { uid = value; }
+
+  CURRENT_DEFAULT_CONSTRUCTOR(User) {}
+
+  using T_BRIEF = User;
```
Comment (mzhurovich) [inline convert_old_db/new_schema.h]: I suppose we should do something with this mandatory `T_BRIEF` using :-/

File: convert_old_db/gen_golden_db.cc (hunk 2 above)
Comment (dkorolev) [inline convert_old_db/gen_golden_db.cc:64]: чил зе фак аут, бро

File: convert_old_db/new_schema.h (hunk 3 above)
Comment (dkorolev) [inline convert_old_db/new_schema.h]: :+1:
76abedc and https://github.com/c5t/Current/commit/e9e6af2102369e131b0da906960657f6636b68a0 (sorry, the latter went into an already open pull request in C5T/Current).

File: convert_old_db/convert.cc (hunk 1 above)
Comment (dkorolev) [inline convert_old_db/convert.cc]: :+1: 0933b14 + https://github.com/c5t/Current/commit/921793eb13e34d0f79e4bd2a2308982a764b4ee8

**Comment by mzhurovich** [comment]
LGTM!

**Pull request merged**